worlds/sc2/mission_order/structs.py

```diff
@@ -527,11 +568,23 @@ def get_missions(self) -> List[SC2MOGenMission]:
     def get_exits(self) -> List[SC2MOGenMission]:
         return self.exits
     
-    def get_visual_requirement(self) -> Union[str, SC2MOGenMission]:
-        return self.get_visual_name()
+    def get_visual_requirement(self, searcher: MissionOrderNode) -> Union[str, SC2MOGenMission]:
```

Comment:
What's a searcher supposed to be? Maybe a `start_node`? I'm not really able to give good naming suggestions because I don't really understand what the naming intent was.

"Searcher" sounds like an object-oriented thing where they make a dedicated object to wrap a function that does some kind o search operation, but I just see you calling `get_parent()` on it which doesn't seem to match that pattern.

Reply:
A searcher is a node that's searching through other nodes. More accurately this node is the node that's currently converting its entry rule dict into an entry rule data structure, which I couldn't think of a better name for. `start_node` is good enough and probably clearer, so I'll use that instead.